documentation/projects/proposals/search_relevancy_sandbox/20230518-implementation_plan_staging_index_rapid_iteration.md

```diff
@@ -0,0 +1,380 @@
+generate a new index in Elasticsearch without requiring a data refresh,
+ingestion server deployment, or any interaction with the ingestion server.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+Once the implementation is complete, a maintainer should be able to run this DAG
+to create a new index on either elasticsearch environment with an altered
+configuration. The maintainer should only need to specify the pieces of the
+configuration that are different from the default configuration, although they
+can supply a full configuration if they wish. The DAG will report when it is
+complete and what the name of the new index is.
+
+## Dependencies
+
+The following are prerequisites for the DAG:
+
+- Installation of the
+  [`elasticsearch` Airflow provider package](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/)
+- Airflow Connections for each Elasticsearch cluster
+
+### Elasticsearch provider
+
+The
+[`elasticsearch` provider](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/)
+will need to be added to the list of dependencies for our catalog Docker image.
+This will be done first and deployed on its own, as the connections will be
+needed for the rest of the work.
+
+The
+[production requirements](https://github.com/WordPress/openverse/blob/3fcce5ade2165955db5bbcb4f679257b3260547b/catalog/requirements_prod.txt#L5)
+will need to be changed from `apache-airflow[amazon,postgres,http]` to
+`apache-airflow[amazon,postgres,http,elasticsearch]`.
+
+### Airflow connections
+
+Once the provider is installed,
+[Airflow Connections](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/connections/elasticsearch.html)
+will need to be created for each Elasticsearch cluster. The connections will be
+named `elasticsearch_production` and `elasticsearch_staging` and reference each
+respective environment. These will be created by hand in the Airflow UI. For
+local development, defaults will be added to the
+[`env.template` file](https://github.com/WordPress/openverse/blob/3fcce5ade2165955db5bbcb4f679257b3260547b/catalog/env.template).
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+No infrastructure changes should be necessary for this implementation, beyond
+the addition of the `elasticsearch` provider package and the Airflow connections
+described above.
+
+### Other projects or work
+
+<!-- Note any projects this plan is dependent on. -->
+
+This work does not depend on any other projects and can be implemented at any
+time. In order to be utilized however, a query parameter on the API will need to
+be used for determining which index to query. This has been implemented in
+specifying which index to use for a given query. This has been implemented in
+#2073.
```

Comment:
```suggestion
be used for determining which index to query. This has been implemented in
#2073.
```

Small editorial change, looks like the sentence was accidentally semi-duplicated.